packages/dev/addons/src/navigation/common/convert.ts

```diff
@@ -0,0 +1,53 @@
+import type { IVector3Like } from "core/Maths/math.like";
+import { Vector3 } from "core/Maths/math.vector";
+import { Logger } from "core/Misc/logger";
+
+import type { ComputeSmoothPathResult } from "../types";
+
+/**
+ *  Converts navigation path points to a Vector3 array.
+ *  @param navPath The navigation path containing points and success status.
+ *  @returns An array of Vector3 points representing the navigation path.
+ */
+export function ConvertNavPathPoints(
```

Comment:
Why is the extra type here needed for navPath? Why isn't it always just `ComputeSmoothPathResult`, which seems almost identical. If it is really needed, maybe add a comment explaining.